hibernate-core/src/main/antlr/org/hibernate/grammars/hql/HqlParser.g4

```diff
@@ -1070,7 +1071,18 @@ anyFunction
  * The `listagg()` ordered set-aggregate function
  */
 listaggFunction
-	: LISTAGG LEFT_PAREN DISTINCT? expressionOrPredicate COMMA expressionOrPredicate onOverflowClause? RIGHT_PAREN withinGroupClause? filterClause?
+	: LISTAGG LEFT_PAREN DISTINCT? expressionOrPredicate COMMA expressionOrPredicate onOverflowClause? RIGHT_PAREN withinGroupClause? filterClause? overClause?
+	;
+
+/**
+ * Special value access window functions
+ */
+valueAccessWindowFunctions
+	: LAG LEFT_PAREN expressionOrPredicate (COMMA expression (COMMA expressionOrPredicate)?)? RIGHT_PAREN nullsClause? filterClause? overClause		# LagFunction
+	| LEAD LEFT_PAREN expressionOrPredicate (COMMA expression (COMMA expressionOrPredicate)?)? RIGHT_PAREN nullsClause? filterClause? overClause	# LeadFunction
+	| FIRST_VALUE LEFT_PAREN expressionOrPredicate RIGHT_PAREN nullsClause? filterClause? overClause 												# FirstValueFunction
+	| LAST_VALUE LEFT_PAREN expressionOrPredicate RIGHT_PAREN nullsClause? filterClause? overClause													# LastValueFunction
+	| NTH_VALUE LEFT_PAREN expressionOrPredicate COMMA expression RIGHT_PAREN nthSideClause? nullsClause? filterClause? overClause 					# NthValueFunction
```

Comment:
Actually, this is a good idea and I'll implement that. Note though that window functions are already dialect-extensible. The `overClause` rule was added to the `genericFunction` rule as well. I added these special cases here because they additionally have the `nullsClause` and `nthSideClause` which only these window functions have, but we can handle that in the code.